src/structures/ClientUser.js

```diff
@@ -46,16 +46,18 @@ class ClientUser extends Structures.get('User') {
     return this.client.presence;
   }
 
-  edit(data) {
-    return this.client.api
-      .users('@me')
-      .patch({ data })
-      .then(newData => {
-        this.client.token = newData.token;
-        const { updated } = this.client.actions.UserUpdate.handle(newData);
-        if (updated) return updated;
-        return this;
-      });
+  /**
+   * Edits the logged in client.
+   * @param {Object} data The new data
+   * @param {string} [data.username] The new username
+   * @param {BufferResolvable|Base64Resolvable} [data.avatar] The new avatar
+   */
+  async edit(data) {
+    const newData = await this.client.api.users('@me').patch({ data });
```

Comment:
```suggestion
    const newData = await this.client.api.users['@me'].patch({ data });
```

Reply:
Both of these are valid uses... We should probably stick to a consistent pattern tho

Reply:
see #5256 🎉

Reply:
@vladfrangu see #5256 🎉